code/modules/surgery/organs/organ_internal.dm

```diff
@@ -362,6 +363,37 @@
 	owner.SetEyeBlurry(0)
 	owner.SetEyeBlind(0)
 
+/obj/item/organ/internal/robotize()
+	if(!(status & ORGAN_ROBOT)) //Don't override the icons for the already-mechanical IPC organs.
+		var/icon/temp = icon('icons/obj/surgery.dmi') //Insensitive to specially-defined icon files for species like the Drask or whomever else. Everyone gets the same robotic heart.
+		if(slot == "heart" && ("[slot]-prosthetic-on" in temp.IconStates()) && ("[slot]-prosthetic-off" in temp.IconStates())) //Give the robotic heart its robotic heart icons if they exist.
+			var/obj/item/organ/internal/heart/H = src
+			H.icon_base = "[slot]-prosthetic"
+			H.dead_icon = "[slot]-prosthetic-off"
+			H.update_icon()
+		else if("[slot]-prosthetic" in temp.IconStates()) //Give the robotic organ its robotic organ icons if they exist.
+			icon_state = "[slot]-prosthetic"
+		name = "mechanical [slot]"
+	..() //Go apply all the organ flags/robotic statuses.
+
+/obj/item/organ/internal/eyes/robotize()
+	colourmatrix = list() //Full colour vision, though this is made redundant by the hardcode in the get_screen_colour() proc from code\modules\mob\mob_helpers.dm.
+	dark_view = 2
+	..()
+	owner.update_client_colour(0) /*Update their colour view if the eyes got a matrix attached to them.
+								Since both mechassisted and mechanical eyes give darknessview of 2 and full colour vision atm, just having this here is fine.*/
+
+/obj/item/organ/internal/mechassist()
+	..()
+	var/icon/temp = icon(icon) //Sensitive to specially-defined icon files since the organs are not fully synthetic.
```

Comment:
`var/list/states = icon_states(icon)`
Then, below...
`("whatever-icon-state" in states)`